plugin-mainmenu/actionview.cpp

```diff
@@ -275,6 +282,40 @@ QSize ActionView::minimumSizeHint() const
     return QSize{0, 0};
 }
 
+void ActionView::mousePressEvent(QMouseEvent* event)
+{
+    if (event->button() == Qt::LeftButton)
+        mDragStartPosition = event->pos();
+
+    QListView::mousePressEvent(event);
+}
+
+void ActionView::mouseMoveEvent(QMouseEvent *event)
+{
+    if (!(event->buttons() & Qt::LeftButton))
+        return;
+
+    if ((event->pos() - mDragStartPosition).manhattanLength() < QApplication::startDragDistance())
+        return;
+
+    XdgAction *a = qobject_cast<XdgAction*>(indexAt(mDragStartPosition).data(ActionView::ActionRole).value<QAction*>());
+    if (!a)
+        return;
+
+    QList<QUrl> urls;
+    urls << QUrl::fromLocalFile(a->desktopFile().fileName());
+
+    QMimeData *mimeData = new QMimeData();
+    mimeData->setUrls(urls);
+
+    QPointer<QDrag> drag = new QDrag(this);
+    drag->setMimeData(mimeData);
+    drag->exec(Qt::CopyAction | Qt::LinkAction);
+    emit requestShowHideMenu();
+    event->accept();
+    drag->deleteLater();
```

Comment:
@slidinghotdog 
Oh, sorry, this is my fault :( Not having paid enough attention to the code, I overlooked the fact that the drop is performed by *another app*, so that deleting `QDrag` cancels dropping (there's no mimetype to drop). 

Please reverse it to your previous code, using an ordinary pointer and NOT deleting `drag`!

The code is good otherwise. When you make the above-mentioned change, I'll merge it.

Reply:
ok, done. it's weird that it worked on quicklaunch and vscode, but not on desktop.